tests/handlers/test_oidc.py

```diff
@@ -159,11 +134,11 @@ def default_config(self) -> Dict[str, Any]:
         return config
 
     def make_homeserver(self, reactor: MemoryReactor, clock: Clock) -> HomeServer:
-        self.http_client = Mock(spec=["get_json"])
-        self.http_client.get_json.side_effect = get_json
-        self.http_client.user_agent = b"Synapse Test"
+        self.fake_provider = FakeOidcServer(clock=clock, issuer=ISSUER)
```

Comment:
```suggestion
        self.fake_server = FakeOidcServer(clock=clock, issuer=ISSUER)
```

(and update the callers)